### PR TITLE
TrackHelper using exceptions instead of returning null.

### PR DESCRIPTION
**Pull request by d4rken**
PR regarding discussion in #163 

> Builder pattern should not require a null-check by default, use exception where necessary.

### CODE REVIEW

**Comment by d4rken** [comment]
I wouldn't mind. Either `3.0.0` or `2.1.0` would be fine with me.

Is it a breaking change though? The nullchecks become superfluos, but nothing previously guaranteed that `TrackHelper` methods don't throw exceptions.

**Comment by d4rken** [comment]
Need to move the exception checking to the build command and then check if we can just try catch exceptions within the trackhelper to achieve something similar to the previous behavior (current one in 2.0) where bad data doesn't necessarily lead to a crash.

File: piwik-sdk/src/main/java/org/piwik/sdk/extra/TrackHelper.java [hunk 1]
```diff
@@ -55,16 +54,39 @@ TrackMe getBaseTrackMe() {
...
+            return safelyWith(piwikApplication.getTracker());
+        }
+
+        /**
+         * {@link #build()} can throw an exception on illegal arguments.
+         * This can be used to avoid crashes when using dynamic {@link TrackMe} arguments.
+         *
+         * @return false if an error occured, true if the TrackMe has been submitted to be dispatched.
+         */
+        public boolean safelyWith(Tracker tracker) {
```
Comment (d4rken) [inline piwik-sdk/src/main/java/org/piwik/sdk/extra/TrackHelper.java]: @dotsbb What do you think about this?

Comment (dotsbb) [inline piwik-sdk/src/main/java/org/piwik/sdk/extra/TrackHelper.java]: Good one!

File: piwik-sdk/build.gradle [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apply plugin: 'com.android.library'
 
-def versionMajor = 2
```
Comment (dotsbb) [inline piwik-sdk/build.gradle]: :+1:

**Pull request merged**